src/Table/TablePagination.js

```diff
@@ -147,12 +153,14 @@ class TablePagination extends React.Component<ProvidedProps & Props> {
       ...other
     } = this.props;
 
-    if (other.component == null || other.component === 'td') {
-      other.colSpan = other.colSpan || 9001; // col-span over everything
+    const more = {};
+
+    if ((component == null || component === 'td') && !colSpan) {
```

Comment:
@oliviertassinari While cleaner than my logic, this seems to be a bit broken: `colSpan` is not used below

Reply:
Fixed.